kore/src/Kore/Parser/Lexeme.hs

```diff
@@ -217,36 +229,44 @@ idCharSet :: CharSet
 idCharSet =
     CharSet.join idFirstCharSet (CharSet.makeCharSet idOtherChars)
 
-{-|'objectIdRawParser' extracts the string representing an @object-identifier@.
-Does not consume whitespace.
--}
-objectIdRawParser :: IdKeywordParsing -> Parser String
-objectIdRawParser keywordParsing = do
-    backslash <- Parser.string "\\" <|> pure ""
-    name <- objectNonslashIdRawParser keywordParsing
-    pure (backslash ++ name)
-
-objectNonslashIdRawParser :: IdKeywordParsing -> Parser String
-objectNonslashIdRawParser =
+symbolIdRawParser :: Parser String
+symbolIdRawParser = do
+    c <- peekChar'
+    if c == '\\'
+    then do
+        skipChar '\\'
+        (c :) <$> genericKoreIdRawParser KeywordsPermitted
+    else genericKoreIdRawParser KeywordsForbidden
+
+-- | Parses a 'Symbol' 'Id'
+symbolIdParser :: Parser Id
+symbolIdParser = genericIdParser symbolIdRawParser
+
+genericKoreIdRawParser :: IdKeywordParsing -> Parser String
+genericKoreIdRawParser =
     genericIdRawParser idFirstCharSet idCharSet
 
+genericKoreIdParser :: Parser Id
+genericKoreIdParser = genericIdParser (genericKoreIdRawParser KeywordsForbidden)
+
 {-|'setVarIdRawParser' extracts the string representing a
 @set-variable-identifier@. Does not consume whitespace.
 
 Always starts with @@@
 -}
 setVarIdRawParser :: Parser String
 setVarIdRawParser = do
-    c <- Parser.char '@'
-    c' <- peekChar'
-    case c' of
-        '`' -> do
-            skipChar c'
-            idToken <- objectNonslashIdRawParser KeywordsPermitted
-            return (c:c':idToken)
-        _ -> do
-            idToken <- objectIdRawParser KeywordsPermitted
-            return (c:idToken)
+    start <- Parser.char '@'
+    end <- genericKoreIdRawParser KeywordsPermitted
+    return (start:end)
+
+{-|'elemVarIdRawParser' extracts the string representing a
+@set-variable-identifier@. Does not consume whitespace.
+
+Always starts with @@@
```

Comment:
Should't @ be escaped, see, e.g., line 121 (documentation for `variableIdParser`)?

Reply:
probably.